### PR TITLE
Add copy assignment op test

### PR DESCRIPTION
**Pull request by geyyer**

### CODE REVIEW

File: test/data_type/test_custom_type.cpp
Comment (andriy-ca) [inline test/data_type/test_custom_type.cpp:1]: Just to be sure, I would destroy or overwrite `right_vec` somehow after the assignment.

**Review by andriy-ca — CHANGES_REQUESTED** [review]

File: test/data_type/test_custom_type.cpp
Comment (geyyer) [reply test/data_type/test_custom_type.cpp:1]: Thanks, good point!

Comment (geyyer) [reply test/data_type/test_custom_type.cpp:1]: Added overwriting with a vector of 0s in [8f4a46c](https://github.com/ROCm/composable_kernel/pull/1718/commits/8f4a46c111d1c84fe10bb02b472cb96551da2549).

**Review by andriy-ca — APPROVED** [review]
Looks Great!

**Pull request merged**